src/com/oltpbenchmark/benchmarks/tpcc/TPCCLoader.java

```diff
@@ -67,19 +69,24 @@ public TPCCLoader(TPCCBenchmark benchmark) {
         }
     }
 
+    private boolean deferConstraintChecks;
     private int numWarehouses = 0;
     private static final int FIRST_UNPROCESSED_O_ID = 2101;
 
     @Override
     public List<LoaderThread> createLoaderThreads() throws SQLException {
         List<LoaderThread> threads = new ArrayList<LoaderThread>();
         final CountDownLatch itemLatch = new CountDownLatch(1);
+        final CountDownLatch warehouseLatch = new CountDownLatch(numWarehouses);
 
         // ITEM
         // This will be invoked first and executed in a single thread.
         threads.add(new LoaderThread() {
             @Override
             public void load(Connection conn) throws SQLException {
+                if (workConf.getDeferConstraintChecks()) {
+                    DisableForeignKeyConstraints(conn);
```

Comment:
Instead of adding constraints, dropping them and then adding them again, it's better to move foreign key statements out of `tpcc-postgres-ddl.sql`.
If `addFkAfterLoad` is true, then call `AddForeignKeyConstraints`, otherwise call the method after loading data.